### PR TITLE
 [perf] minor adjustments to `store#push` flow

### PR DESCRIPTION
**Pull request by trentmwillis**
Some simple suggestions of refactoring to squeeze out a little extra performance when calling `store#push`. In our application this shows ~50ms improvement for slower devices on medium-sized payloads. Changes are as follows:

- Don't call `coerceId` in `_load` since it gets called in `_internalModelForId`
- Use `Ember.isArray` instead of `Ember.typeOf`
- Use `for-loop`s instead of `Array.prototype` methods

### CODE REVIEW

**Comment by stefanpenner** [comment]
We should prefer Array.isArray over Ember.isArray as the later is merely an alias for the former and exists for compar

File: packages/ember-data/lib/system/store.js [hunk 1]
```diff
@@ -1672,13 +1672,21 @@ Store = Service.extend({
...
+      for (i = 0, l = included.length; i < l; i++) {
+        this._pushInternalModel(included[i]);
+      }
     }
 
-    if (Ember.typeOf(data.data) === 'array') {
-      var internalModels = data.data.map((recordData) => this._pushInternalModel(recordData));
-      return internalModels.map((internalModel) => internalModel.getRecord());
+    if (isArray(data.data)) {
+      var internalModels = [];
```
Comment (stefanpenner) [inline packages/ember-data/lib/system/store.js]: new Array(length)

**Comment by stefanpenner** [comment]
Two comments otherwise this Lgtm

**Comment by trentmwillis** [comment]
@stefanpenner updated, looks like AppVeyor had issues with installing node modules though

**Comment by stefanpenner** [comment]
@trentmwillis silly appveyor...

anyways thanks this looks good to me.  Its really sad that `forEach` & `map` have such a silly high cost..

**Pull request merged**